client/polling/poll.go

```diff
@@ -0,0 +1,190 @@
+package polling
+
+import (
+	"context"
+	"errors"
+	abci "github.com/tendermint/tendermint/abci/types"
+	"github.com/tendermint/tendermint/libs/log"
+	"github.com/tendermint/tendermint/rpc/client"
+	coretypes "github.com/tendermint/tendermint/rpc/core/types"
+	"math"
+	"math/rand"
+	"strings"
+	"time"
+)
+
+const (
+	magicForBlockDoesNotExistErr = "must be less than or equal to the current blockchain height"
+	baseDelayBetweenTransactions = 25 * time.Millisecond
+	maxDelayBetweenTransactions  = 2500 * time.Millisecond
+)
+
+var (
+	errCapacityInvalid = errors.New("capacity must be 1 or greater")
+	errNoBlocks        = errors.New("rpc call for latest block returned nil")
+)
+
+func PollForBlocks(ctx context.Context, logger log.Logger, c client.Client, outCapacity ...int) (out <-chan abci.ResponseDeliverTx, err error) {
+	logger = logger.With("cmp", "block-poller")
+
+	// Create an outgoing channel for events
+	outCap := 1
+	if len(outCapacity) > 0 {
+		outCap = outCapacity[0]
+		if outCap < 1 {
+			return nil, errCapacityInvalid
+		}
+	}
+	outc := make(chan abci.ResponseDeliverTx, outCap)
+
+	// Get the current height
+	block, err := c.Block(ctx, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	if block == nil || block.Block == nil {
+		return nil, errNoBlocks
+	}
+
+	height := block.Block.Height
+
+	// Run until context dies
+	go func() {
```

Comment:
`defer close(outc)`?

Reply:
I'm not listening for close events, but it should be safe to close it since only 1 goroutine is going to be writing to this. Will add.

Reply:
it's so the listener can use a `for` loop.